addons/httprox/common/src/main/java/org/commonjava/indy/httprox/util/HttpConduitWrapper.java

```diff
@@ -112,6 +113,12 @@ public void writeStatus( final int code, final String message )
         sinkChannel.write( b );
     }
 
+    private void writeClose()
+                    throws IOException
+    {
+        writeHeader( "Connection", "close\r\n" );
```

Comment:
Will this happen even if we've written content already?

Reply:
I see, this is only ever called on 404

Reply:
Yeah. For now, it is only for 404 case. When I tested with some client ( curl or wget ), I found the client hung if we didn't send the additional \r\n in this 404 case. After adding it, all clients are happy.